### PR TITLE
weightings with factor added

### PR DESCRIPTION
**Pull request by kandelirina**
https://bringg.atlassian.net/secure/RapidBoard.jspa?rapidView=4&projectKey=OPTI&modal=detail&selectedIssue=OPTI-130

### CODE REVIEW

File: core/src/main/java/com/graphhopper/GraphHopper.java [hunk 1]
```diff
@@ -1068,6 +1077,7 @@ else if (ALT_ROUTE.equalsIgnoreCase(algoStr))
 
                 AlgorithmOptions algoOpts = AlgorithmOptions.start().
                         algorithm(algoStr).traversalMode(tMode).weighting(weighting).
+                        weightingsFactorGetter(weightFactorsGetter).
```
Comment (irisshainsky) [inline core/src/main/java/com/graphhopper/GraphHopper.java]: really confusing name

File: core/src/main/java/com/graphhopper/GraphHopperAPI.java [hunk 2]
```diff
@@ -17,6 +17,8 @@
  */
 package com.graphhopper;
 
+import com.graphhopper.routing.WeightFactorsGetter;
```
Comment (irisshainsky) [inline core/src/main/java/com/graphhopper/GraphHopperAPI.java]: should be removed unnecessary import

File: core/src/main/java/com/graphhopper/routing/WeightFactorsGetter.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.graphhopper.routing;
+
+import com.graphhopper.util.EdgeIteratorState;
+
+public interface WeightFactorsGetter {
+    double getFactor(EdgeIteratorState edgeState, boolean reverse);
```
Comment (irisshainsky) [inline core/src/main/java/com/graphhopper/routing/WeightFactorsGetter.java]: should be chaned to edgeState without reverse, will count on edgeId, base edge and adj edge

Comment (kandelirina) [reply core/src/main/java/com/graphhopper/routing/WeightFactorsGetter.java]: the factor retrieved by adjNode->baseNode if reverse or baseNode->adjNode, so we need to know if it's reverse or not

File: core/src/main/java/com/graphhopper/routing/WeightFactorsGetter.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.graphhopper.routing;
+
+import com.graphhopper.util.EdgeIteratorState;
+
+public interface WeightFactorsGetter {
```
Comment (kobyb) [inline core/src/main/java/com/graphhopper/routing/WeightFactorsGetter.java]: rename to WeightFactor. getter is not a class name, it's a function name

File: core/src/main/java/com/graphhopper/routing/weighting/WeightingsWithFactors.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package com.graphhopper.routing.weighting;
+
+import com.graphhopper.routing.WeightFactorsGetter;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.HintsMap;
+import com.graphhopper.util.EdgeIteratorState;
+
+public class WeightingsWithFactors implements Weighting {
```
Comment (kobyb) [inline core/src/main/java/com/graphhopper/routing/weighting/WeightingsWithFactors.java]: rename to FactoredWeightings

**Pull request merged**